components/agents/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/oauth2/SAML2GrantManager.java

```diff
@@ -79,7 +84,7 @@ protected String executePost(String urlParameters, String basicAuthHeader)
             connection.setRequestProperty("Authorization", "Basic " + basicAuthHeader);
 
             connection.setRequestProperty("Content-Length", "" +
-                    Integer.toString(urlParameters.getBytes().length));
+                    Integer.toString(urlParameters.getBytes(Charset.forName("UTF-8")).length));
```

Comment:
can use StandardCharsets.UTF_8 instead of Charset.forName

(from java.nio.charset package )